{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "cd87559e-ae5a-4d5c-806b-518ac494e059",
   "metadata": {},
   "source": [
    "# Установка пакетов"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ea4b5c6f-ed27-40a9-b972-7b63c4e973f7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: rdkit in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (2023.9.6)\n",
      "Requirement already satisfied: numpy in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from rdkit) (1.26.4)\n",
      "Requirement already satisfied: Pillow in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from rdkit) (11.1.0)\n",
      "\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m24.0\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m25.2\u001b[0m\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49m/home/kostanew/anaconda3/envs/p4env/bin/python -m pip install --upgrade pip\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "# `rdkit` -- основной пакет для работы со SMILES\n",
    "%pip install rdkit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5870c97e-a5bb-43b7-896e-75b0602ddb10",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pybel in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (0.15.5)\n",
      "Requirement already satisfied: networkx>=2.4 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pybel) (3.2.1)\n",
      "Requirement already satisfied: sqlalchemy in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pybel) (2.0.25)\n",
      "Requirement already satisfied: click in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pybel) (8.1.7)\n",
      "Requirement already satisfied: click-plugins in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pybel) (1.1.1.2)\n",
      "Requirement already satisfied: bel-resources>=0.0.3 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pybel) (0.0.3)\n",
      "Requirement already satisfied: more-itertools in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pybel) (10.8.0)\n",
      "Requirement already satisfied: requests in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pybel) (2.32.3)\n",
      "Requirement already satisfied: requests-file in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pybel) (2.1.0)\n",
      "Requirement already satisfied: pyparsing in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pybel) (3.1.1)\n",
      "Requirement already satisfied: tqdm in /home/kostanew/.local/lib/python3.10/site-packages (from pybel) (4.65.0)\n",
      "Requirement already satisfied: humanize in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pybel) (4.13.0)\n",
      "Requirement already satisfied: tabulate in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pybel) (0.9.0)\n",
      "Requirement already satisfied: pandas in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pybel) (2.1.4)\n",
      "Requirement already satisfied: jsonschema in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pybel) (4.20.0)\n",
      "Requirement already satisfied: bioregistry in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pybel) (0.12.38)\n",
      "Requirement already satisfied: ratelimit in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pybel) (2.2.1)\n",
      "Requirement already satisfied: pystow>=0.1.2 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pybel) (0.7.8)\n",
      "Requirement already satisfied: psycopg2-binary in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pybel) (2.9.10)\n",
      "Requirement already satisfied: multisplitby in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from bel-resources>=0.0.3->pybel) (0.0.1)\n",
      "Requirement already satisfied: typing-extensions in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pystow>=0.1.2->pybel) (4.13.2)\n",
      "Requirement already satisfied: more_click>=0.1.2 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from bioregistry->pybel) (0.1.2)\n",
      "Requirement already satisfied: pydantic>=2.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pydantic[email]>=2.0->bioregistry->pybel) (2.5.3)\n",
      "Requirement already satisfied: curies>=0.7.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from bioregistry->pybel) (0.10.23)\n",
      "Requirement already satisfied: attrs>=22.2.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from jsonschema->pybel) (23.2.0)\n",
      "Requirement already satisfied: jsonschema-specifications>=2023.03.6 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from jsonschema->pybel) (2023.12.1)\n",
      "Requirement already satisfied: referencing>=0.28.4 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from jsonschema->pybel) (0.32.1)\n",
      "Requirement already satisfied: rpds-py>=0.7.1 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from jsonschema->pybel) (0.16.2)\n",
      "Requirement already satisfied: numpy<2,>=1.22.4 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pandas->pybel) (1.26.4)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pandas->pybel) (2.8.2)\n",
      "Requirement already satisfied: pytz>=2020.1 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pandas->pybel) (2023.3.post1)\n",
      "Requirement already satisfied: tzdata>=2022.1 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pandas->pybel) (2023.4)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from requests->pybel) (3.3.2)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from requests->pybel) (3.7)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from requests->pybel) (2.2.1)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from requests->pybel) (2024.6.2)\n",
      "Requirement already satisfied: greenlet!=0.4.17 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from sqlalchemy->pybel) (3.0.3)\n",
      "Requirement already satisfied: pytrie in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from curies>=0.7.0->bioregistry->pybel) (0.4.0)\n",
      "Requirement already satisfied: annotated-types>=0.4.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pydantic>=2.0->pydantic[email]>=2.0->bioregistry->pybel) (0.6.0)\n",
      "Requirement already satisfied: pydantic-core==2.14.6 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pydantic>=2.0->pydantic[email]>=2.0->bioregistry->pybel) (2.14.6)\n",
      "Requirement already satisfied: email-validator>=2.0.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pydantic[email]>=2.0->bioregistry->pybel) (2.3.0)\n",
      "Requirement already satisfied: six>=1.5 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from python-dateutil>=2.8.2->pandas->pybel) (1.16.0)\n",
      "Requirement already satisfied: dnspython>=2.0.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from email-validator>=2.0.0->pydantic[email]>=2.0->bioregistry->pybel) (2.8.0)\n",
      "Requirement already satisfied: sortedcontainers in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pytrie->curies>=0.7.0->bioregistry->pybel) (2.4.0)\n",
      "\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m24.0\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m25.2\u001b[0m\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49m/home/kostanew/anaconda3/envs/p4env/bin/python -m pip install --upgrade pip\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "# `pybel` -- Python-обёртка над OpenBabel\n",
    "# %pip install openbabel-wheel # Нужно раскомментировать и установить тоже в случае появления ошибок.\n",
    "%pip install pybel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e7abd65e-4f02-4286-99d7-875f163a4462",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting xgboost\n",
      "  Downloading xgboost-3.0.5-py3-none-manylinux_2_28_x86_64.whl.metadata (2.1 kB)\n",
      "Requirement already satisfied: numpy in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from xgboost) (1.26.4)\n",
      "Requirement already satisfied: nvidia-nccl-cu12 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from xgboost) (2.19.3)\n",
      "Requirement already satisfied: scipy in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from xgboost) (1.11.4)\n",
      "Downloading xgboost-3.0.5-py3-none-manylinux_2_28_x86_64.whl (94.9 MB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m94.9/94.9 MB\u001b[0m \u001b[31m21.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hInstalling collected packages: xgboost\n",
      "Successfully installed xgboost-3.0.5\n",
      "\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m24.0\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m25.2\u001b[0m\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49m/home/kostanew/anaconda3/envs/p4env/bin/python -m pip install --upgrade pip\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "# XGBoost -- классификаторы и регрессоры для табличных данных\n",
    "%pip install xgboost"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "10cdd4e9-6857-4ea6-a025-5c6af29410b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting xtb\n",
      "  Downloading xtb-22.1-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl.metadata (4.6 kB)\n",
      "Requirement already satisfied: cffi in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from xtb) (1.16.0)\n",
      "Requirement already satisfied: numpy in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from xtb) (1.26.4)\n",
      "Requirement already satisfied: pycparser in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from cffi->xtb) (2.21)\n",
      "Downloading xtb-22.1-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (17.4 MB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m17.4/17.4 MB\u001b[0m \u001b[31m12.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hInstalling collected packages: xtb\n",
      "Successfully installed xtb-22.1\n",
      "\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m24.0\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m25.2\u001b[0m\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49m/home/kostanew/anaconda3/envs/p4env/bin/python -m pip install --upgrade pip\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "%pip install xtb # Оптимизация конформаций с помощью полуэмпирического потенциала"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "54073aa6-d404-4239-a4c5-0112cc490201",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pyscf in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (2.7.0)\n",
      "Requirement already satisfied: numpy!=1.16,!=1.17,>=1.13 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pyscf) (1.26.4)\n",
      "Requirement already satisfied: scipy>=1.6.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pyscf) (1.11.4)\n",
      "Requirement already satisfied: h5py>=2.7 in /home/kostanew/.local/lib/python3.10/site-packages (from pyscf) (3.10.0)\n",
      "Requirement already satisfied: setuptools in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pyscf) (69.5.1)\n",
      "\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m24.0\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m25.2\u001b[0m\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49m/home/kostanew/anaconda3/envs/p4env/bin/python -m pip install --upgrade pip\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "%pip install pyscf # Библиотека для DFT расчётов"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e7903f74-683d-4d58-8c7b-742de969b347",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pyberny in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (0.6.3)\n",
      "Requirement already satisfied: numpy<2.0,>=1.15 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pyberny) (1.26.4)\n",
      "\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m24.0\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m25.2\u001b[0m\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49m/home/kostanew/anaconda3/envs/p4env/bin/python -m pip install --upgrade pip\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "%pip install pyberny # Оптимизация геометрий для pyscf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "82c3f497-b183-40ea-b184-37ad9713b1c7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: mace-torch in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (0.3.10)\n",
      "Collecting mace-models\n",
      "  Downloading mace_models-0.1.6-py3-none-any.whl.metadata (5.4 kB)\n",
      "Requirement already satisfied: torch>=1.12 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from mace-torch) (2.2.0)\n",
      "Requirement already satisfied: e3nn==0.4.4 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from mace-torch) (0.4.4)\n",
      "Requirement already satisfied: numpy in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from mace-torch) (1.26.4)\n",
      "Requirement already satisfied: opt_einsum in /home/kostanew/.local/lib/python3.10/site-packages (from mace-torch) (3.3.0)\n",
      "Requirement already satisfied: ase in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from mace-torch) (3.22.1)\n",
      "Requirement already satisfied: torch-ema in /home/kostanew/.local/lib/python3.10/site-packages (from mace-torch) (0.3)\n",
      "Requirement already satisfied: prettytable in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from mace-torch) (3.15.1)\n",
      "Requirement already satisfied: matscipy in /home/kostanew/.local/lib/python3.10/site-packages (from mace-torch) (1.0.0)\n",
      "Requirement already satisfied: h5py in /home/kostanew/.local/lib/python3.10/site-packages (from mace-torch) (3.10.0)\n",
      "Requirement already satisfied: torchmetrics in /home/kostanew/.local/lib/python3.10/site-packages (from mace-torch) (1.0.1)\n",
      "Requirement already satisfied: python-hostlist in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from mace-torch) (2.2.1)\n",
      "Requirement already satisfied: configargparse in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from mace-torch) (1.7)\n",
      "Requirement already satisfied: GitPython in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from mace-torch) (3.1.43)\n",
      "Requirement already satisfied: pyYAML in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from mace-torch) (6.0.1)\n",
      "Requirement already satisfied: tqdm in /home/kostanew/.local/lib/python3.10/site-packages (from mace-torch) (4.65.0)\n",
      "Requirement already satisfied: matplotlib in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from mace-torch) (3.10.1)\n",
      "Requirement already satisfied: pandas in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from mace-torch) (2.1.4)\n",
      "Requirement already satisfied: sympy in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from e3nn==0.4.4->mace-torch) (1.12)\n",
      "Requirement already satisfied: scipy in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from e3nn==0.4.4->mace-torch) (1.11.4)\n",
      "Requirement already satisfied: opt-einsum-fx>=0.1.4 in /home/kostanew/.local/lib/python3.10/site-packages (from e3nn==0.4.4->mace-torch) (0.1.4)\n",
      "Collecting dvc-s3<4.0,>=3.2 (from mace-models)\n",
      "  Downloading dvc_s3-3.2.2-py3-none-any.whl.metadata (1.5 kB)\n",
      "Collecting zntrack<0.9.0,>=0.8.1 (from mace-models)\n",
      "  Downloading zntrack-0.8.9-py3-none-any.whl.metadata (11 kB)\n",
      "Collecting dvc (from dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading dvc-3.63.0-py3-none-any.whl.metadata (17 kB)\n",
      "Collecting s3fs>=2024.12.0 (from dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading s3fs-2025.9.0-py3-none-any.whl.metadata (1.4 kB)\n",
      "Collecting aiobotocore>=2.5.0 (from aiobotocore[boto3]>=2.5.0->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading aiobotocore-2.24.2-py3-none-any.whl.metadata (25 kB)\n",
      "Collecting flatten_dict<1,>=0.4.1 (from dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading flatten_dict-0.4.2-py2.py3-none-any.whl.metadata (9.2 kB)\n",
      "Requirement already satisfied: contourpy>=1.0.1 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from matplotlib->mace-torch) (1.2.0)\n",
      "Requirement already satisfied: cycler>=0.10 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from matplotlib->mace-torch) (0.12.1)\n",
      "Requirement already satisfied: fonttools>=4.22.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from matplotlib->mace-torch) (4.47.0)\n",
      "Requirement already satisfied: kiwisolver>=1.3.1 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from matplotlib->mace-torch) (1.4.5)\n",
      "Requirement already satisfied: packaging>=20.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from matplotlib->mace-torch) (23.2)\n",
      "Requirement already satisfied: pillow>=8 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from matplotlib->mace-torch) (11.1.0)\n",
      "Requirement already satisfied: pyparsing>=2.3.1 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from matplotlib->mace-torch) (3.1.1)\n",
      "Requirement already satisfied: python-dateutil>=2.7 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from matplotlib->mace-torch) (2.8.2)\n",
      "Requirement already satisfied: filelock in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torch>=1.12->mace-torch) (3.13.1)\n",
      "Requirement already satisfied: typing-extensions>=4.8.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torch>=1.12->mace-torch) (4.13.2)\n",
      "Requirement already satisfied: networkx in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torch>=1.12->mace-torch) (3.2.1)\n",
      "Requirement already satisfied: jinja2 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torch>=1.12->mace-torch) (3.1.3)\n",
      "Requirement already satisfied: fsspec in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torch>=1.12->mace-torch) (2023.12.2)\n",
      "Requirement already satisfied: nvidia-cuda-nvrtc-cu12==12.1.105 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torch>=1.12->mace-torch) (12.1.105)\n",
      "Requirement already satisfied: nvidia-cuda-runtime-cu12==12.1.105 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torch>=1.12->mace-torch) (12.1.105)\n",
      "Requirement already satisfied: nvidia-cuda-cupti-cu12==12.1.105 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torch>=1.12->mace-torch) (12.1.105)\n",
      "Requirement already satisfied: nvidia-cudnn-cu12==8.9.2.26 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torch>=1.12->mace-torch) (8.9.2.26)\n",
      "Requirement already satisfied: nvidia-cublas-cu12==12.1.3.1 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torch>=1.12->mace-torch) (12.1.3.1)\n",
      "Requirement already satisfied: nvidia-cufft-cu12==11.0.2.54 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torch>=1.12->mace-torch) (11.0.2.54)\n",
      "Requirement already satisfied: nvidia-curand-cu12==10.3.2.106 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torch>=1.12->mace-torch) (10.3.2.106)\n",
      "Requirement already satisfied: nvidia-cusolver-cu12==11.4.5.107 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torch>=1.12->mace-torch) (11.4.5.107)\n",
      "Requirement already satisfied: nvidia-cusparse-cu12==12.1.0.106 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torch>=1.12->mace-torch) (12.1.0.106)\n",
      "Requirement already satisfied: nvidia-nccl-cu12==2.19.3 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torch>=1.12->mace-torch) (2.19.3)\n",
      "Requirement already satisfied: nvidia-nvtx-cu12==12.1.105 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torch>=1.12->mace-torch) (12.1.105)\n",
      "Requirement already satisfied: triton==2.2.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torch>=1.12->mace-torch) (2.2.0)\n",
      "Requirement already satisfied: nvidia-nvjitlink-cu12 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from nvidia-cusolver-cu12==11.4.5.107->torch>=1.12->mace-torch) (12.3.101)\n",
      "Collecting pandas (from mace-torch)\n",
      "  Downloading pandas-2.3.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (91 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m91.2/91.2 kB\u001b[0m \u001b[31m1.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0ma \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hCollecting pyYAML (from mace-torch)\n",
      "  Using cached PyYAML-6.0.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.1 kB)\n",
      "Collecting tqdm (from mace-torch)\n",
      "  Using cached tqdm-4.67.1-py3-none-any.whl.metadata (57 kB)\n",
      "Requirement already satisfied: typer>=0.15.1 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from zntrack<0.9.0,>=0.8.1->mace-models) (0.15.2)\n",
      "Collecting znfields>=0.1.2 (from zntrack<0.9.0,>=0.8.1->mace-models)\n",
      "  Downloading znfields-0.1.2-py3-none-any.whl.metadata (1.6 kB)\n",
      "Collecting znflow>=0.2.5 (from zntrack<0.9.0,>=0.8.1->mace-models)\n",
      "  Downloading znflow-0.2.5-py3-none-any.whl.metadata (7.3 kB)\n",
      "Collecting znjson>=0.2.6 (from zntrack<0.9.0,>=0.8.1->mace-models)\n",
      "  Downloading znjson-0.2.6-py3-none-any.whl.metadata (3.5 kB)\n",
      "Requirement already satisfied: pytz>=2020.1 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pandas->mace-torch) (2023.3.post1)\n",
      "Requirement already satisfied: tzdata>=2022.7 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pandas->mace-torch) (2023.4)\n",
      "Requirement already satisfied: gitdb<5,>=4.0.1 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from GitPython->mace-torch) (4.0.11)\n",
      "Requirement already satisfied: looseversion in /home/kostanew/.local/lib/python3.10/site-packages (from matscipy->mace-torch) (1.3.0)\n",
      "Requirement already satisfied: wcwidth in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from prettytable->mace-torch) (0.2.13)\n",
      "Requirement already satisfied: lightning-utilities>=0.7.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from torchmetrics->mace-torch) (0.14.0)\n",
      "Collecting aiohttp<4.0.0,>=3.9.2 (from aiobotocore>=2.5.0->aiobotocore[boto3]>=2.5.0->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading aiohttp-3.12.15-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.7 kB)\n",
      "Collecting aioitertools<1.0.0,>=0.5.1 (from aiobotocore>=2.5.0->aiobotocore[boto3]>=2.5.0->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Using cached aioitertools-0.12.0-py3-none-any.whl.metadata (3.8 kB)\n",
      "Collecting botocore<1.40.19,>=1.40.15 (from aiobotocore>=2.5.0->aiobotocore[boto3]>=2.5.0->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading botocore-1.40.18-py3-none-any.whl.metadata (5.7 kB)\n",
      "Requirement already satisfied: jmespath<2.0.0,>=0.7.1 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from aiobotocore>=2.5.0->aiobotocore[boto3]>=2.5.0->dvc-s3<4.0,>=3.2->mace-models) (1.0.1)\n",
      "Requirement already satisfied: multidict<7.0.0,>=6.0.0 in /home/kostanew/.local/lib/python3.10/site-packages (from aiobotocore>=2.5.0->aiobotocore[boto3]>=2.5.0->dvc-s3<4.0,>=3.2->mace-models) (6.0.4)\n",
      "Requirement already satisfied: wrapt<2.0.0,>=1.10.10 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from aiobotocore>=2.5.0->aiobotocore[boto3]>=2.5.0->dvc-s3<4.0,>=3.2->mace-models) (1.17.2)\n",
      "Collecting boto3<1.40.19,>=1.40.15 (from aiobotocore[boto3]>=2.5.0->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading boto3-1.40.18-py3-none-any.whl.metadata (6.7 kB)\n",
      "Requirement already satisfied: attrs>=22.2.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from dvc->dvc-s3<4.0,>=3.2->mace-models) (23.2.0)\n",
      "Collecting celery (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading celery-5.5.3-py3-none-any.whl.metadata (22 kB)\n",
      "Requirement already satisfied: colorama>=0.3.9 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from dvc->dvc-s3<4.0,>=3.2->mace-models) (0.4.6)\n",
      "Collecting configobj>=5.0.9 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading configobj-5.0.9-py2.py3-none-any.whl.metadata (3.2 kB)\n",
      "Requirement already satisfied: distro>=1.3 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from dvc->dvc-s3<4.0,>=3.2->mace-models) (1.9.0)\n",
      "Collecting dpath<3,>=2.1.0 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading dpath-2.2.0-py3-none-any.whl.metadata (15 kB)\n",
      "Collecting dulwich (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading dulwich-0.24.1-cp310-cp310-manylinux_2_28_x86_64.whl.metadata (5.2 kB)\n",
      "Collecting dvc-data<3.17,>=3.16.2 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading dvc_data-3.16.12-py3-none-any.whl.metadata (5.1 kB)\n",
      "Collecting dvc-http>=2.29.0 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading dvc_http-2.32.0-py3-none-any.whl.metadata (1.3 kB)\n",
      "Collecting dvc-objects (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading dvc_objects-5.1.1-py3-none-any.whl.metadata (3.8 kB)\n",
      "Collecting dvc-render<2,>=1.0.1 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading dvc_render-1.0.2-py3-none-any.whl.metadata (5.4 kB)\n",
      "Collecting dvc-studio-client<1,>=0.21 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading dvc_studio_client-0.22.0-py3-none-any.whl.metadata (4.4 kB)\n",
      "Collecting dvc-task<1,>=0.3.0 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading dvc_task-0.40.2-py3-none-any.whl.metadata (10.0 kB)\n",
      "Collecting flufl.lock<9,>=8.1.0 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading flufl_lock-8.2.0-py3-none-any.whl.metadata (3.3 kB)\n",
      "Collecting fsspec (from torch>=1.12->mace-torch)\n",
      "  Downloading fsspec-2025.9.0-py3-none-any.whl.metadata (10 kB)\n",
      "Collecting funcy>=1.14 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading funcy-2.0-py2.py3-none-any.whl.metadata (5.9 kB)\n",
      "Collecting grandalf<1,>=0.7 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading grandalf-0.8-py3-none-any.whl.metadata (1.7 kB)\n",
      "Collecting gto<2,>=1.6.0 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading gto-1.8.0-py3-none-any.whl.metadata (4.7 kB)\n",
      "Requirement already satisfied: hydra-core>=1.1 in /home/kostanew/.local/lib/python3.10/site-packages (from dvc->dvc-s3<4.0,>=3.2->mace-models) (1.2.0)\n",
      "Collecting iterative-telemetry>=0.0.7 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading iterative_telemetry-0.0.10-py3-none-any.whl.metadata (4.1 kB)\n",
      "Collecting kombu (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading kombu-5.5.4-py3-none-any.whl.metadata (3.5 kB)\n",
      "Requirement already satisfied: omegaconf in /home/kostanew/.local/lib/python3.10/site-packages (from dvc->dvc-s3<4.0,>=3.2->mace-models) (2.3.0)\n",
      "Collecting pathspec>=0.10.3 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Using cached pathspec-0.12.1-py3-none-any.whl.metadata (21 kB)\n",
      "Requirement already satisfied: platformdirs<5,>=3.1.1 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from dvc->dvc-s3<4.0,>=3.2->mace-models) (4.1.0)\n",
      "Requirement already satisfied: psutil>=5.8 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from dvc->dvc-s3<4.0,>=3.2->mace-models) (5.9.8)\n",
      "Requirement already satisfied: pydot>=1.2.4 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from dvc->dvc-s3<4.0,>=3.2->mace-models) (3.0.3)\n",
      "Collecting pygtrie>=2.3.2 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading pygtrie-2.5.0-py3-none-any.whl.metadata (7.5 kB)\n",
      "Requirement already satisfied: requests>=2.22 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from dvc->dvc-s3<4.0,>=3.2->mace-models) (2.32.3)\n",
      "Requirement already satisfied: rich>=12 in /home/kostanew/.local/lib/python3.10/site-packages (from dvc->dvc-s3<4.0,>=3.2->mace-models) (13.7.1)\n",
      "Requirement already satisfied: ruamel.yaml>=0.17.11 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from dvc->dvc-s3<4.0,>=3.2->mace-models) (0.18.10)\n",
      "Collecting scmrepo<4,>=3.5.2 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading scmrepo-3.5.2-py3-none-any.whl.metadata (4.8 kB)\n",
      "Collecting shortuuid>=0.5 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading shortuuid-1.0.13-py3-none-any.whl.metadata (5.8 kB)\n",
      "Collecting shtab<2,>=1.3.4 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading shtab-1.7.2-py3-none-any.whl.metadata (7.4 kB)\n",
      "Requirement already satisfied: tabulate>=0.8.7 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from dvc->dvc-s3<4.0,>=3.2->mace-models) (0.9.0)\n",
      "Collecting tomlkit>=0.11.1 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Using cached tomlkit-0.13.3-py3-none-any.whl.metadata (2.8 kB)\n",
      "Collecting voluptuous>=0.11.7 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading voluptuous-0.15.2-py3-none-any.whl.metadata (20 kB)\n",
      "Collecting zc.lockfile>=1.2.1 (from dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading zc.lockfile-3.0.post1-py3-none-any.whl.metadata (6.2 kB)\n",
      "Requirement already satisfied: six<2.0,>=1.12 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from flatten_dict<1,>=0.4.1->dvc-s3<4.0,>=3.2->mace-models) (1.16.0)\n",
      "Requirement already satisfied: smmap<6,>=3.0.1 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from gitdb<5,>=4.0.1->GitPython->mace-torch) (5.0.0)\n",
      "Requirement already satisfied: setuptools in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from lightning-utilities>=0.7.0->torchmetrics->mace-torch) (69.5.1)\n",
      "Requirement already satisfied: click>=8.0.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from typer>=0.15.1->zntrack<0.9.0,>=0.8.1->mace-models) (8.1.7)\n",
      "Requirement already satisfied: shellingham>=1.3.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from typer>=0.15.1->zntrack<0.9.0,>=0.8.1->mace-models) (1.5.4)\n",
      "Collecting networkx (from torch>=1.12->mace-torch)\n",
      "  Using cached networkx-3.4.2-py3-none-any.whl.metadata (6.3 kB)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from jinja2->torch>=1.12->mace-torch) (2.1.3)\n",
      "Requirement already satisfied: mpmath>=0.19 in /home/kostanew/.local/lib/python3.10/site-packages (from sympy->e3nn==0.4.4->mace-torch) (1.3.0)\n",
      "Collecting aiohappyeyeballs>=2.5.0 (from aiohttp<4.0.0,>=3.9.2->aiobotocore>=2.5.0->aiobotocore[boto3]>=2.5.0->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Using cached aiohappyeyeballs-2.6.1-py3-none-any.whl.metadata (5.9 kB)\n",
      "Collecting aiosignal>=1.4.0 (from aiohttp<4.0.0,>=3.9.2->aiobotocore>=2.5.0->aiobotocore[boto3]>=2.5.0->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading aiosignal-1.4.0-py3-none-any.whl.metadata (3.7 kB)\n",
      "Requirement already satisfied: async-timeout<6.0,>=4.0 in /home/kostanew/.local/lib/python3.10/site-packages (from aiohttp<4.0.0,>=3.9.2->aiobotocore>=2.5.0->aiobotocore[boto3]>=2.5.0->dvc-s3<4.0,>=3.2->mace-models) (4.0.2)\n",
      "Requirement already satisfied: frozenlist>=1.1.1 in /home/kostanew/.local/lib/python3.10/site-packages (from aiohttp<4.0.0,>=3.9.2->aiobotocore>=2.5.0->aiobotocore[boto3]>=2.5.0->dvc-s3<4.0,>=3.2->mace-models) (1.3.3)\n",
      "Collecting propcache>=0.2.0 (from aiohttp<4.0.0,>=3.9.2->aiobotocore>=2.5.0->aiobotocore[boto3]>=2.5.0->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading propcache-0.3.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (12 kB)\n",
      "Collecting yarl<2.0,>=1.17.0 (from aiohttp<4.0.0,>=3.9.2->aiobotocore>=2.5.0->aiobotocore[boto3]>=2.5.0->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading yarl-1.20.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (73 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m73.9/73.9 kB\u001b[0m \u001b[31m3.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hCollecting s3transfer<0.14.0,>=0.13.0 (from boto3<1.40.19,>=1.40.15->aiobotocore[boto3]>=2.5.0->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading s3transfer-0.13.1-py3-none-any.whl.metadata (1.7 kB)\n",
      "Requirement already satisfied: urllib3!=2.2.0,<3,>=1.25.4 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from botocore<1.40.19,>=1.40.15->aiobotocore>=2.5.0->aiobotocore[boto3]>=2.5.0->dvc-s3<4.0,>=3.2->mace-models) (2.2.1)\n",
      "Collecting dictdiffer>=0.8.1 (from dvc-data<3.17,>=3.16.2->dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading dictdiffer-0.9.0-py2.py3-none-any.whl.metadata (4.8 kB)\n",
      "Collecting diskcache>=5.2.1 (from dvc-data<3.17,>=3.16.2->dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Using cached diskcache-5.6.3-py3-none-any.whl.metadata (20 kB)\n",
      "Collecting sqltrie<1,>=0.11.0 (from dvc-data<3.17,>=3.16.2->dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading sqltrie-0.11.2-py3-none-any.whl.metadata (3.3 kB)\n",
      "Collecting orjson<4,>=3 (from dvc-data<3.17,>=3.16.2->dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading orjson-3.11.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (41 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m41.9/41.9 kB\u001b[0m \u001b[31m3.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hCollecting aiohttp-retry>=2.5.0 (from dvc-http>=2.29.0->dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading aiohttp_retry-2.9.1-py3-none-any.whl.metadata (8.8 kB)\n",
      "Collecting billiard<5.0,>=4.2.1 (from celery->dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading billiard-4.2.1-py3-none-any.whl.metadata (4.4 kB)\n",
      "Collecting vine<6.0,>=5.1.0 (from celery->dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading vine-5.1.0-py3-none-any.whl.metadata (2.7 kB)\n",
      "Collecting click-didyoumean>=0.3.0 (from celery->dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading click_didyoumean-0.3.1-py3-none-any.whl.metadata (3.9 kB)\n",
      "Collecting click-repl>=0.2.0 (from celery->dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading click_repl-0.3.0-py3-none-any.whl.metadata (3.6 kB)\n",
      "Requirement already satisfied: click-plugins>=1.1.1 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from celery->dvc->dvc-s3<4.0,>=3.2->mace-models) (1.1.1.2)\n",
      "Collecting atpublic (from flufl.lock<9,>=8.1.0->dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading atpublic-6.0.1-py3-none-any.whl.metadata (2.0 kB)\n",
      "Requirement already satisfied: entrypoints in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from gto<2,>=1.6.0->dvc->dvc-s3<4.0,>=3.2->mace-models) (0.4)\n",
      "Requirement already satisfied: pydantic!=2.0.0,<3,>=1.9.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from gto<2,>=1.6.0->dvc->dvc-s3<4.0,>=3.2->mace-models) (2.5.3)\n",
      "Collecting semver>=2.13.0 (from gto<2,>=1.6.0->dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading semver-3.0.4-py3-none-any.whl.metadata (6.8 kB)\n",
      "Requirement already satisfied: antlr4-python3-runtime==4.9.* in /home/kostanew/.local/lib/python3.10/site-packages (from hydra-core>=1.1->dvc->dvc-s3<4.0,>=3.2->mace-models) (4.9.3)\n",
      "Requirement already satisfied: appdirs in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from iterative-telemetry>=0.0.7->dvc->dvc-s3<4.0,>=3.2->mace-models) (1.4.4)\n",
      "Collecting amqp<6.0.0,>=5.1.1 (from kombu->dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading amqp-5.3.1-py3-none-any.whl.metadata (8.9 kB)\n",
      "Collecting tzdata>=2022.7 (from pandas->mace-torch)\n",
      "  Using cached tzdata-2025.2-py2.py3-none-any.whl.metadata (1.4 kB)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from requests>=2.22->dvc->dvc-s3<4.0,>=3.2->mace-models) (3.3.2)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from requests>=2.22->dvc->dvc-s3<4.0,>=3.2->mace-models) (3.7)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from requests>=2.22->dvc->dvc-s3<4.0,>=3.2->mace-models) (2024.6.2)\n",
      "Requirement already satisfied: markdown-it-py>=2.2.0 in /home/kostanew/.local/lib/python3.10/site-packages (from rich>=12->dvc->dvc-s3<4.0,>=3.2->mace-models) (3.0.0)\n",
      "Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /home/kostanew/.local/lib/python3.10/site-packages (from rich>=12->dvc->dvc-s3<4.0,>=3.2->mace-models) (2.17.2)\n",
      "Requirement already satisfied: ruamel.yaml.clib>=0.2.7 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from ruamel.yaml>=0.17.11->dvc->dvc-s3<4.0,>=3.2->mace-models) (0.2.12)\n",
      "Collecting pygit2>=1.14.0 (from scmrepo<4,>=3.5.2->dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading pygit2-1.18.2-cp310-cp310-manylinux_2_26_x86_64.manylinux_2_28_x86_64.whl.metadata (3.6 kB)\n",
      "Collecting asyncssh<3,>=2.13.1 (from scmrepo<4,>=3.5.2->dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading asyncssh-2.21.0-py3-none-any.whl.metadata (9.9 kB)\n",
      "Collecting cryptography>=39.0 (from asyncssh<3,>=2.13.1->scmrepo<4,>=3.5.2->dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading cryptography-45.0.7-cp37-abi3-manylinux_2_34_x86_64.whl.metadata (5.7 kB)\n",
      "Requirement already satisfied: prompt-toolkit>=3.0.36 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from click-repl>=0.2.0->celery->dvc->dvc-s3<4.0,>=3.2->mace-models) (3.0.42)\n",
      "Requirement already satisfied: mdurl~=0.1 in /home/kostanew/.local/lib/python3.10/site-packages (from markdown-it-py>=2.2.0->rich>=12->dvc->dvc-s3<4.0,>=3.2->mace-models) (0.1.2)\n",
      "Requirement already satisfied: annotated-types>=0.4.0 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pydantic!=2.0.0,<3,>=1.9.0->gto<2,>=1.6.0->dvc->dvc-s3<4.0,>=3.2->mace-models) (0.6.0)\n",
      "Requirement already satisfied: pydantic-core==2.14.6 in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from pydantic!=2.0.0,<3,>=1.9.0->gto<2,>=1.6.0->dvc->dvc-s3<4.0,>=3.2->mace-models) (2.14.6)\n",
      "Collecting cffi>=1.17.0 (from pygit2>=1.14.0->scmrepo<4,>=3.5.2->dvc->dvc-s3<4.0,>=3.2->mace-models)\n",
      "  Downloading cffi-2.0.0-cp310-cp310-manylinux2014_x86_64.manylinux_2_17_x86_64.whl.metadata (2.6 kB)\n",
      "Requirement already satisfied: pycparser in /home/kostanew/anaconda3/envs/p4env/lib/python3.10/site-packages (from cffi>=1.17.0->pygit2>=1.14.0->scmrepo<4,>=3.5.2->dvc->dvc-s3<4.0,>=3.2->mace-models) (2.21)\n",
      "Downloading mace_models-0.1.6-py3-none-any.whl (5.1 kB)\n",
      "Downloading dvc_s3-3.2.2-py3-none-any.whl (13 kB)\n",
      "Downloading zntrack-0.8.9-py3-none-any.whl (68 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m68.3/68.3 kB\u001b[0m \u001b[31m6.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading pandas-2.3.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.3 MB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m12.3/12.3 MB\u001b[0m \u001b[31m30.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hUsing cached PyYAML-6.0.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (751 kB)\n",
      "Using cached tqdm-4.67.1-py3-none-any.whl (78 kB)\n",
      "Downloading aiobotocore-2.24.2-py3-none-any.whl (85 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m85.4/85.4 kB\u001b[0m \u001b[31m26.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading dvc-3.63.0-py3-none-any.whl (466 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m466.2/466.2 kB\u001b[0m \u001b[31m42.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading flatten_dict-0.4.2-py2.py3-none-any.whl (9.7 kB)\n",
      "Downloading fsspec-2025.9.0-py3-none-any.whl (199 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m199.3/199.3 kB\u001b[0m \u001b[31m35.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading s3fs-2025.9.0-py3-none-any.whl (30 kB)\n",
      "Downloading znfields-0.1.2-py3-none-any.whl (7.7 kB)\n",
      "Downloading znflow-0.2.5-py3-none-any.whl (23 kB)\n",
      "Using cached networkx-3.4.2-py3-none-any.whl (1.7 MB)\n",
      "Downloading znjson-0.2.6-py3-none-any.whl (15 kB)\n",
      "Downloading aiohttp-3.12.15-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.6 MB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.6/1.6 MB\u001b[0m \u001b[31m20.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0ma \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hDownloading aioitertools-0.12.0-py3-none-any.whl (24 kB)\n",
      "Downloading boto3-1.40.18-py3-none-any.whl (140 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m140.1/140.1 kB\u001b[0m \u001b[31m29.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading botocore-1.40.18-py3-none-any.whl (14.0 MB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m14.0/14.0 MB\u001b[0m \u001b[31m26.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hDownloading configobj-5.0.9-py2.py3-none-any.whl (35 kB)\n",
      "Downloading dpath-2.2.0-py3-none-any.whl (17 kB)\n",
      "Downloading dvc_data-3.16.12-py3-none-any.whl (78 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m78.2/78.2 kB\u001b[0m \u001b[31m24.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading dvc_http-2.32.0-py3-none-any.whl (12 kB)\n",
      "Downloading dvc_objects-5.1.1-py3-none-any.whl (33 kB)\n",
      "Downloading dvc_render-1.0.2-py3-none-any.whl (22 kB)\n",
      "Downloading dvc_studio_client-0.22.0-py3-none-any.whl (16 kB)\n",
      "Downloading dvc_task-0.40.2-py3-none-any.whl (21 kB)\n",
      "Downloading celery-5.5.3-py3-none-any.whl (438 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m438.8/438.8 kB\u001b[0m \u001b[31m37.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading flufl_lock-8.2.0-py3-none-any.whl (11 kB)\n",
      "Downloading funcy-2.0-py2.py3-none-any.whl (30 kB)\n",
      "Downloading grandalf-0.8-py3-none-any.whl (41 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m41.8/41.8 kB\u001b[0m \u001b[31m12.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading gto-1.8.0-py3-none-any.whl (45 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m45.4/45.4 kB\u001b[0m \u001b[31m11.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading iterative_telemetry-0.0.10-py3-none-any.whl (10 kB)\n",
      "Downloading kombu-5.5.4-py3-none-any.whl (210 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m210.0/210.0 kB\u001b[0m \u001b[31m35.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading vine-5.1.0-py3-none-any.whl (9.6 kB)\n",
      "Using cached tzdata-2025.2-py2.py3-none-any.whl (347 kB)\n",
      "Using cached pathspec-0.12.1-py3-none-any.whl (31 kB)\n",
      "Downloading pygtrie-2.5.0-py3-none-any.whl (25 kB)\n",
      "Downloading scmrepo-3.5.2-py3-none-any.whl (73 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m73.9/73.9 kB\u001b[0m \u001b[31m24.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading dulwich-0.24.1-cp310-cp310-manylinux_2_28_x86_64.whl (1.2 MB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.2/1.2 MB\u001b[0m \u001b[31m39.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading shortuuid-1.0.13-py3-none-any.whl (10 kB)\n",
      "Downloading shtab-1.7.2-py3-none-any.whl (14 kB)\n",
      "Using cached tomlkit-0.13.3-py3-none-any.whl (38 kB)\n",
      "Downloading voluptuous-0.15.2-py3-none-any.whl (31 kB)\n",
      "Downloading zc.lockfile-3.0.post1-py3-none-any.whl (9.8 kB)\n",
      "Using cached aiohappyeyeballs-2.6.1-py3-none-any.whl (15 kB)\n",
      "Downloading aiohttp_retry-2.9.1-py3-none-any.whl (10.0 kB)\n",
      "Downloading aiosignal-1.4.0-py3-none-any.whl (7.5 kB)\n",
      "Downloading amqp-5.3.1-py3-none-any.whl (50 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m50.9/50.9 kB\u001b[0m \u001b[31m15.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading asyncssh-2.21.0-py3-none-any.whl (374 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m374.9/374.9 kB\u001b[0m \u001b[31m35.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading billiard-4.2.1-py3-none-any.whl (86 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m86.8/86.8 kB\u001b[0m \u001b[31m26.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading click_didyoumean-0.3.1-py3-none-any.whl (3.6 kB)\n",
      "Downloading click_repl-0.3.0-py3-none-any.whl (10 kB)\n",
      "Downloading dictdiffer-0.9.0-py2.py3-none-any.whl (16 kB)\n",
      "Downloading diskcache-5.6.3-py3-none-any.whl (45 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m45.5/45.5 kB\u001b[0m \u001b[31m13.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading orjson-3.11.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (132 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m133.0/133.0 kB\u001b[0m \u001b[31m17.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading propcache-0.3.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (198 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m198.3/198.3 kB\u001b[0m \u001b[31m32.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading pygit2-1.18.2-cp310-cp310-manylinux_2_26_x86_64.manylinux_2_28_x86_64.whl (5.5 MB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m5.5/5.5 MB\u001b[0m \u001b[31m31.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hDownloading s3transfer-0.13.1-py3-none-any.whl (85 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m85.3/85.3 kB\u001b[0m \u001b[31m19.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading semver-3.0.4-py3-none-any.whl (17 kB)\n",
      "Downloading sqltrie-0.11.2-py3-none-any.whl (17 kB)\n",
      "Downloading yarl-1.20.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (326 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m326.1/326.1 kB\u001b[0m \u001b[31m28.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading atpublic-6.0.1-py3-none-any.whl (6.4 kB)\n",
      "Downloading cffi-2.0.0-cp310-cp310-manylinux2014_x86_64.manylinux_2_17_x86_64.whl (216 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m216.5/216.5 kB\u001b[0m \u001b[31m34.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading cryptography-45.0.7-cp37-abi3-manylinux_2_34_x86_64.whl (4.4 MB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m4.4/4.4 MB\u001b[0m \u001b[31m31.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hInstalling collected packages: pygtrie, funcy, dictdiffer, znjson, znfields, zc.lockfile, voluptuous, vine, tzdata, tqdm, tomlkit, shtab, shortuuid, semver, pyYAML, propcache, pathspec, orjson, networkx, grandalf, fsspec, flatten_dict, dvc-render, dulwich, dpath, diskcache, configobj, click-didyoumean, cffi, billiard, atpublic, aiosignal, aioitertools, aiohappyeyeballs, znflow, yarl, sqltrie, pygit2, pandas, iterative-telemetry, flufl.lock, dvc-studio-client, dvc-objects, cryptography, click-repl, botocore, amqp, s3transfer, kombu, dvc-data, asyncssh, aiohttp, celery, boto3, aiohttp-retry, aiobotocore, scmrepo, s3fs, dvc-task, dvc-http, gto, dvc, zntrack, dvc-s3, mace-models\n",
      "  Attempting uninstall: tzdata\n",
      "    Found existing installation: tzdata 2023.4\n",
      "    Uninstalling tzdata-2023.4:\n",
      "      Successfully uninstalled tzdata-2023.4\n",
      "  Attempting uninstall: tqdm\n",
      "    Found existing installation: tqdm 4.65.0\n",
      "    Uninstalling tqdm-4.65.0:\n",
      "      Successfully uninstalled tqdm-4.65.0\n",
      "  Attempting uninstall: pyYAML\n",
      "    Found existing installation: PyYAML 6.0.1\n",
      "    Uninstalling PyYAML-6.0.1:\n",
      "      Successfully uninstalled PyYAML-6.0.1\n",
      "  Attempting uninstall: networkx\n",
      "    Found existing installation: networkx 3.2.1\n",
      "    Uninstalling networkx-3.2.1:\n",
      "      Successfully uninstalled networkx-3.2.1\n",
      "  Attempting uninstall: fsspec\n",
      "    Found existing installation: fsspec 2023.12.2\n",
      "    Uninstalling fsspec-2023.12.2:\n",
      "      Successfully uninstalled fsspec-2023.12.2\n",
      "  Attempting uninstall: cffi\n",
      "    Found existing installation: cffi 1.16.0\n",
      "    Uninstalling cffi-1.16.0:\n",
      "      Successfully uninstalled cffi-1.16.0\n",
      "  Attempting uninstall: aiosignal\n",
      "    Found existing installation: aiosignal 1.3.1\n",
      "    Uninstalling aiosignal-1.3.1:\n",
      "      Successfully uninstalled aiosignal-1.3.1\n",
      "  Attempting uninstall: yarl\n",
      "    Found existing installation: yarl 1.8.2\n",
      "    Uninstalling yarl-1.8.2:\n",
      "      Successfully uninstalled yarl-1.8.2\n",
      "  Attempting uninstall: pandas\n",
      "    Found existing installation: pandas 2.1.4\n",
      "    Uninstalling pandas-2.1.4:\n",
      "      Successfully uninstalled pandas-2.1.4\n",
      "  Attempting uninstall: botocore\n",
      "    Found existing installation: botocore 1.34.92\n",
      "    Uninstalling botocore-1.34.92:\n",
      "      Successfully uninstalled botocore-1.34.92\n",
      "  Attempting uninstall: s3transfer\n",
      "    Found existing installation: s3transfer 0.10.1\n",
      "    Uninstalling s3transfer-0.10.1:\n",
      "      Successfully uninstalled s3transfer-0.10.1\n",
      "  Attempting uninstall: aiohttp\n",
      "    Found existing installation: aiohttp 3.8.4\n",
      "    Uninstalling aiohttp-3.8.4:\n",
      "      Successfully uninstalled aiohttp-3.8.4\n",
      "  Attempting uninstall: boto3\n",
      "    Found existing installation: boto3 1.34.92\n",
      "    Uninstalling boto3-1.34.92:\n",
      "      Successfully uninstalled boto3-1.34.92\n",
      "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "lightning 2.1.4 requires fsspec[http]<2025.0,>=2022.5.0, but you have fsspec 2025.9.0 which is incompatible.\n",
      "nabladft 2.0.0a0 requires e3nn==0.5.1, but you have e3nn 0.4.4 which is incompatible.\n",
      "torch-harmonics 0.7.3 requires torch>=2.4.0, but you have torch 2.2.0 which is incompatible.\u001b[0m\u001b[31m\n",
      "\u001b[0mSuccessfully installed aiobotocore-2.24.2 aiohappyeyeballs-2.6.1 aiohttp-3.12.15 aiohttp-retry-2.9.1 aioitertools-0.12.0 aiosignal-1.4.0 amqp-5.3.1 asyncssh-2.21.0 atpublic-6.0.1 billiard-4.2.1 boto3-1.40.18 botocore-1.40.18 celery-5.5.3 cffi-2.0.0 click-didyoumean-0.3.1 click-repl-0.3.0 configobj-5.0.9 cryptography-45.0.7 dictdiffer-0.9.0 diskcache-5.6.3 dpath-2.2.0 dulwich-0.24.1 dvc-3.63.0 dvc-data-3.16.12 dvc-http-2.32.0 dvc-objects-5.1.1 dvc-render-1.0.2 dvc-s3-3.2.2 dvc-studio-client-0.22.0 dvc-task-0.40.2 flatten_dict-0.4.2 flufl.lock-8.2.0 fsspec-2024.12.0 funcy-2.0 grandalf-0.8 gto-1.8.0 iterative-telemetry-0.0.10 kombu-5.5.4 mace-models-0.1.6 networkx-3.4.2 orjson-3.11.3 pandas-2.3.2 pathspec-0.12.1 propcache-0.3.2 pyYAML-6.0.2 pygit2-1.18.2 pygtrie-2.5.0 s3fs-2025.9.0 s3transfer-0.13.1 scmrepo-3.5.2 semver-3.0.4 shortuuid-1.0.13 shtab-1.7.2 sqltrie-0.11.2 tomlkit-0.13.3 tqdm-4.67.1 tzdata-2025.2 vine-5.1.0 voluptuous-0.15.2 yarl-1.20.1 zc.lockfile-3.0.post1 znfields-0.1.2 znflow-0.2.5 znjson-0.2.6 zntrack-0.8.9\n",
      "\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m24.0\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m25.2\u001b[0m\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49m/home/kostanew/anaconda3/envs/p4env/bin/python -m pip install --upgrade pip\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "%pip install mace-torch mace-models"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "18c93a40-67ea-4779-9cf3-e0e247148b40",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "        You're using the MACE-MP-0_small model. The model is released under the MIT license.\n",
      "        Note:\n",
      "        If you are using this model, please cite the relevant paper for the Materials Project,\n",
      "        any paper associated with the MACE model, and also the following:\n",
      "        - MACE-Universal by Yuan Chiang, 2023, Hugging Face, Revision e5ebd9b,\n",
      "            DOI: 10.57967/hf/1202, URL: https://huggingface.co/cyrusyc/mace-universal\n",
      "        - Matbench Discovery by Janosh Riebesell, Rhys EA Goodall, Philipp Benner, Yuan Chiang,\n",
      "            Alpha A Lee, Anubhav Jain, Kristin A Persson, 2023, arXiv:2308.14920\n",
      "        - https://arxiv.org/abs/2401.00096\n",
      "           \n",
      "Default dtype float32 does not match model dtype float64, converting models to float32.\n"
     ]
    }
   ],
   "source": [
    "import mace_models\n",
    "model = mace_models.load(\"MACE-MP-0_small\")\n",
    "calculator = model.get_calculator()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0e70d3c8-d832-4268-82be-f31511cf797d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fatal: destination path 'nablaDFT' already exists and is not an empty directory.\n",
      "Error processing line 1 of /home/kostanew/anaconda3/lib/python3.9/site-packages/distutils-precedence.pth:\n",
      "\n",
      "  Traceback (most recent call last):\n",
      "    File \"/home/kostanew/anaconda3/lib/python3.9/site.py\", line 169, in addpackage\n",
      "      exec(line)\n",
      "    File \"<string>\", line 1, in <module>\n",
      "  ModuleNotFoundError: No module named '_distutils_hack'\n",
      "\n",
      "Remainder of file ignored\n",
      "Processing /home/kostanew/2025/airi-DS4chem/nablaDFT\n",
      "  Installing build dependencies ... \u001b[?25ldone\n",
      "\u001b[?25h  Getting requirements to build wheel ... \u001b[?25ldone\n",
      "\u001b[?25h  Preparing metadata (pyproject.toml) ... \u001b[?25ldone\n",
      "\u001b[?25hCollecting torch-scatter@ https://data.pyg.org/whl/torch-2.2.0%2Bcu121/torch_scatter-2.1.2%2Bpt22cu121-cp39-cp39-linux_x86_64.whl (from nablaDFT==2.0.0a0)\n",
      "  Using cached https://data.pyg.org/whl/torch-2.2.0%2Bcu121/torch_scatter-2.1.2%2Bpt22cu121-cp39-cp39-linux_x86_64.whl (10.9 MB)\n",
      "Collecting torch-sparse@ https://data.pyg.org/whl/torch-2.2.0%2Bcu121/torch_sparse-0.6.18%2Bpt22cu121-cp39-cp39-linux_x86_64.whl (from nablaDFT==2.0.0a0)\n",
      "  Using cached https://data.pyg.org/whl/torch-2.2.0%2Bcu121/torch_sparse-0.6.18%2Bpt22cu121-cp39-cp39-linux_x86_64.whl (5.1 MB)\n",
      "Collecting torch-cluster@ https://data.pyg.org/whl/torch-2.2.0%2Bcu121/torch_cluster-1.6.3%2Bpt22cu121-cp39-cp39-linux_x86_64.whl (from nablaDFT==2.0.0a0)\n",
      "  Using cached https://data.pyg.org/whl/torch-2.2.0%2Bcu121/torch_cluster-1.6.3%2Bpt22cu121-cp39-cp39-linux_x86_64.whl (3.4 MB)\n",
      "Requirement already satisfied: numpy<2.0.0,>=1.26 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from nablaDFT==2.0.0a0) (1.26.4)\n",
      "Collecting sympy==1.12 (from nablaDFT==2.0.0a0)\n",
      "  Using cached sympy-1.12-py3-none-any.whl.metadata (12 kB)\n",
      "Requirement already satisfied: ase==3.22.1 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from nablaDFT==2.0.0a0) (3.22.1)\n",
      "Collecting h5py==3.10.0 (from nablaDFT==2.0.0a0)\n",
      "  Using cached h5py-3.10.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.5 kB)\n",
      "Collecting apsw==3.45.1.0 (from nablaDFT==2.0.0a0)\n",
      "  Using cached apsw-3.45.1.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.4 kB)\n",
      "Requirement already satisfied: schnetpack==2.0.4 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from nablaDFT==2.0.0a0) (2.0.4)\n",
      "Requirement already satisfied: tensorboardX in /home/kostanew/anaconda3/lib/python3.9/site-packages (from nablaDFT==2.0.0a0) (2.6.2.2)\n",
      "Requirement already satisfied: pyyaml in /home/kostanew/anaconda3/lib/python3.9/site-packages (from nablaDFT==2.0.0a0) (6.0)\n",
      "Collecting hydra-core==1.2.0 (from nablaDFT==2.0.0a0)\n",
      "  Using cached hydra_core-1.2.0-py3-none-any.whl.metadata (4.0 kB)\n",
      "Collecting torch==2.2.0 (from nablaDFT==2.0.0a0)\n",
      "  Using cached torch-2.2.0-cp39-cp39-manylinux1_x86_64.whl.metadata (25 kB)\n",
      "Collecting pytorch_lightning==2.1.4 (from nablaDFT==2.0.0a0)\n",
      "  Using cached pytorch_lightning-2.1.4-py3-none-any.whl.metadata (21 kB)\n",
      "Requirement already satisfied: torch-geometric==2.4.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from nablaDFT==2.0.0a0) (2.4.0)\n",
      "Requirement already satisfied: torchmetrics==1.0.1 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from nablaDFT==2.0.0a0) (1.0.1)\n",
      "Requirement already satisfied: hydra-colorlog>=1.1.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from nablaDFT==2.0.0a0) (1.2.0)\n",
      "Requirement already satisfied: rich in /home/kostanew/anaconda3/lib/python3.9/site-packages (from nablaDFT==2.0.0a0) (12.5.1)\n",
      "Requirement already satisfied: fasteners in /home/kostanew/anaconda3/lib/python3.9/site-packages (from nablaDFT==2.0.0a0) (0.19)\n",
      "Requirement already satisfied: dirsync in /home/kostanew/anaconda3/lib/python3.9/site-packages (from nablaDFT==2.0.0a0) (2.2.5)\n",
      "Requirement already satisfied: torch-ema==0.3 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from nablaDFT==2.0.0a0) (0.3)\n",
      "Requirement already satisfied: matscipy in /home/kostanew/anaconda3/lib/python3.9/site-packages (from nablaDFT==2.0.0a0) (0.8.0)\n",
      "Requirement already satisfied: python-dotenv in /home/kostanew/anaconda3/lib/python3.9/site-packages (from nablaDFT==2.0.0a0) (1.0.0)\n",
      "Collecting wandb==0.16.3 (from nablaDFT==2.0.0a0)\n",
      "  Using cached wandb-0.16.3-py3-none-any.whl.metadata (9.9 kB)\n",
      "Collecting e3nn==0.5.1 (from nablaDFT==2.0.0a0)\n",
      "  Using cached e3nn-0.5.1-py3-none-any.whl.metadata (5.7 kB)\n",
      "Requirement already satisfied: matplotlib>=3.1.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from ase==3.22.1->nablaDFT==2.0.0a0) (3.5.1)\n",
      "Requirement already satisfied: scipy>=1.1.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from ase==3.22.1->nablaDFT==2.0.0a0) (1.13.0)\n",
      "Collecting opt-einsum-fx>=0.1.4 (from e3nn==0.5.1->nablaDFT==2.0.0a0)\n",
      "  Using cached opt_einsum_fx-0.1.4-py3-none-any.whl.metadata (3.3 kB)\n",
      "Requirement already satisfied: omegaconf~=2.2 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from hydra-core==1.2.0->nablaDFT==2.0.0a0) (2.3.0)\n",
      "Requirement already satisfied: antlr4-python3-runtime==4.9.* in /home/kostanew/anaconda3/lib/python3.9/site-packages (from hydra-core==1.2.0->nablaDFT==2.0.0a0) (4.9.3)\n",
      "Requirement already satisfied: packaging in /home/kostanew/anaconda3/lib/python3.9/site-packages (from hydra-core==1.2.0->nablaDFT==2.0.0a0) (23.1)\n",
      "Requirement already satisfied: tqdm>=4.57.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from pytorch_lightning==2.1.4->nablaDFT==2.0.0a0) (4.64.0)\n",
      "Requirement already satisfied: fsspec>=2022.5.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from fsspec[http]>=2022.5.0->pytorch_lightning==2.1.4->nablaDFT==2.0.0a0) (2023.3.0)\n",
      "Requirement already satisfied: typing-extensions>=4.0.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from pytorch_lightning==2.1.4->nablaDFT==2.0.0a0) (4.15.0)\n",
      "Requirement already satisfied: lightning-utilities>=0.8.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from pytorch_lightning==2.1.4->nablaDFT==2.0.0a0) (0.10.0)\n",
      "Requirement already satisfied: tensorboard in /home/kostanew/anaconda3/lib/python3.9/site-packages (from schnetpack==2.0.4->nablaDFT==2.0.0a0) (2.12.0)\n",
      "Requirement already satisfied: mpmath>=0.19 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from sympy==1.12->nablaDFT==2.0.0a0) (1.2.1)\n",
      "Requirement already satisfied: filelock in /home/kostanew/anaconda3/lib/python3.9/site-packages (from torch==2.2.0->nablaDFT==2.0.0a0) (3.12.4)\n",
      "Requirement already satisfied: networkx in /home/kostanew/anaconda3/lib/python3.9/site-packages (from torch==2.2.0->nablaDFT==2.0.0a0) (2.7.1)\n",
      "Requirement already satisfied: jinja2 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from torch==2.2.0->nablaDFT==2.0.0a0) (3.0.3)\n",
      "Collecting nvidia-cuda-nvrtc-cu12==12.1.105 (from torch==2.2.0->nablaDFT==2.0.0a0)\n",
      "  Using cached nvidia_cuda_nvrtc_cu12-12.1.105-py3-none-manylinux1_x86_64.whl.metadata (1.5 kB)\n",
      "Collecting nvidia-cuda-runtime-cu12==12.1.105 (from torch==2.2.0->nablaDFT==2.0.0a0)\n",
      "  Using cached nvidia_cuda_runtime_cu12-12.1.105-py3-none-manylinux1_x86_64.whl.metadata (1.5 kB)\n",
      "Collecting nvidia-cuda-cupti-cu12==12.1.105 (from torch==2.2.0->nablaDFT==2.0.0a0)\n",
      "  Using cached nvidia_cuda_cupti_cu12-12.1.105-py3-none-manylinux1_x86_64.whl.metadata (1.6 kB)\n",
      "Collecting nvidia-cudnn-cu12==8.9.2.26 (from torch==2.2.0->nablaDFT==2.0.0a0)\n",
      "  Using cached nvidia_cudnn_cu12-8.9.2.26-py3-none-manylinux1_x86_64.whl.metadata (1.6 kB)\n",
      "Collecting nvidia-cublas-cu12==12.1.3.1 (from torch==2.2.0->nablaDFT==2.0.0a0)\n",
      "  Using cached nvidia_cublas_cu12-12.1.3.1-py3-none-manylinux1_x86_64.whl.metadata (1.5 kB)\n",
      "Collecting nvidia-cufft-cu12==11.0.2.54 (from torch==2.2.0->nablaDFT==2.0.0a0)\n",
      "  Using cached nvidia_cufft_cu12-11.0.2.54-py3-none-manylinux1_x86_64.whl.metadata (1.5 kB)\n",
      "Collecting nvidia-curand-cu12==10.3.2.106 (from torch==2.2.0->nablaDFT==2.0.0a0)\n",
      "  Using cached nvidia_curand_cu12-10.3.2.106-py3-none-manylinux1_x86_64.whl.metadata (1.5 kB)\n",
      "Collecting nvidia-cusolver-cu12==11.4.5.107 (from torch==2.2.0->nablaDFT==2.0.0a0)\n",
      "  Using cached nvidia_cusolver_cu12-11.4.5.107-py3-none-manylinux1_x86_64.whl.metadata (1.6 kB)\n",
      "Collecting nvidia-cusparse-cu12==12.1.0.106 (from torch==2.2.0->nablaDFT==2.0.0a0)\n",
      "  Using cached nvidia_cusparse_cu12-12.1.0.106-py3-none-manylinux1_x86_64.whl.metadata (1.6 kB)\n",
      "Collecting nvidia-nccl-cu12==2.19.3 (from torch==2.2.0->nablaDFT==2.0.0a0)\n",
      "  Using cached nvidia_nccl_cu12-2.19.3-py3-none-manylinux1_x86_64.whl.metadata (1.8 kB)\n",
      "Collecting nvidia-nvtx-cu12==12.1.105 (from torch==2.2.0->nablaDFT==2.0.0a0)\n",
      "  Using cached nvidia_nvtx_cu12-12.1.105-py3-none-manylinux1_x86_64.whl.metadata (1.7 kB)\n",
      "Collecting triton==2.2.0 (from torch==2.2.0->nablaDFT==2.0.0a0)\n",
      "  Using cached triton-2.2.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (1.4 kB)\n",
      "Requirement already satisfied: requests in /home/kostanew/anaconda3/lib/python3.9/site-packages (from torch-geometric==2.4.0->nablaDFT==2.0.0a0) (2.32.5)\n",
      "Requirement already satisfied: pyparsing in /home/kostanew/anaconda3/lib/python3.9/site-packages (from torch-geometric==2.4.0->nablaDFT==2.0.0a0) (3.0.4)\n",
      "Requirement already satisfied: scikit-learn in /home/kostanew/anaconda3/lib/python3.9/site-packages (from torch-geometric==2.4.0->nablaDFT==2.0.0a0) (1.3.0)\n",
      "Requirement already satisfied: psutil>=5.8.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from torch-geometric==2.4.0->nablaDFT==2.0.0a0) (5.8.0)\n",
      "Requirement already satisfied: Click!=8.0.0,>=7.1 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from wandb==0.16.3->nablaDFT==2.0.0a0) (8.0.4)\n",
      "Requirement already satisfied: GitPython!=3.1.29,>=1.0.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from wandb==0.16.3->nablaDFT==2.0.0a0) (3.1.40)\n",
      "Requirement already satisfied: sentry-sdk>=1.0.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from wandb==0.16.3->nablaDFT==2.0.0a0) (1.36.0)\n",
      "Requirement already satisfied: docker-pycreds>=0.4.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from wandb==0.16.3->nablaDFT==2.0.0a0) (0.4.0)\n",
      "Requirement already satisfied: setproctitle in /home/kostanew/anaconda3/lib/python3.9/site-packages (from wandb==0.16.3->nablaDFT==2.0.0a0) (1.3.3)\n",
      "Requirement already satisfied: setuptools in /home/kostanew/anaconda3/lib/python3.9/site-packages (from wandb==0.16.3->nablaDFT==2.0.0a0) (44.1.1)\n",
      "Requirement already satisfied: appdirs>=1.4.3 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from wandb==0.16.3->nablaDFT==2.0.0a0) (1.4.4)\n",
      "Requirement already satisfied: protobuf!=4.21.0,<5,>=3.15.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from wandb==0.16.3->nablaDFT==2.0.0a0) (4.22.1)\n",
      "Collecting nvidia-nvjitlink-cu12 (from nvidia-cusolver-cu12==11.4.5.107->torch==2.2.0->nablaDFT==2.0.0a0)\n",
      "  Using cached nvidia_nvjitlink_cu12-12.9.86-py3-none-manylinux2010_x86_64.manylinux_2_12_x86_64.whl.metadata (1.7 kB)\n",
      "Requirement already satisfied: colorlog in /home/kostanew/anaconda3/lib/python3.9/site-packages (from hydra-colorlog>=1.1.0->nablaDFT==2.0.0a0) (6.7.0)\n",
      "Requirement already satisfied: six in /home/kostanew/anaconda3/lib/python3.9/site-packages (from dirsync->nablaDFT==2.0.0a0) (1.16.0)\n",
      "Requirement already satisfied: commonmark<0.10.0,>=0.9.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from rich->nablaDFT==2.0.0a0) (0.9.1)\n",
      "Requirement already satisfied: pygments<3.0.0,>=2.6.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from rich->nablaDFT==2.0.0a0) (2.11.2)\n",
      "Requirement already satisfied: aiohttp!=4.0.0a0,!=4.0.0a1 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from fsspec[http]>=2022.5.0->pytorch_lightning==2.1.4->nablaDFT==2.0.0a0) (3.8.1)\n",
      "Requirement already satisfied: gitdb<5,>=4.0.1 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from GitPython!=3.1.29,>=1.0.0->wandb==0.16.3->nablaDFT==2.0.0a0) (4.0.11)\n",
      "Requirement already satisfied: cycler>=0.10 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from matplotlib>=3.1.0->ase==3.22.1->nablaDFT==2.0.0a0) (0.11.0)\n",
      "Requirement already satisfied: fonttools>=4.22.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from matplotlib>=3.1.0->ase==3.22.1->nablaDFT==2.0.0a0) (4.25.0)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from matplotlib>=3.1.0->ase==3.22.1->nablaDFT==2.0.0a0) (1.3.2)\n",
      "Requirement already satisfied: pillow>=6.2.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from matplotlib>=3.1.0->ase==3.22.1->nablaDFT==2.0.0a0) (9.0.1)\n",
      "Requirement already satisfied: python-dateutil>=2.7 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from matplotlib>=3.1.0->ase==3.22.1->nablaDFT==2.0.0a0) (2.8.2)\n",
      "Requirement already satisfied: opt-einsum in /home/kostanew/anaconda3/lib/python3.9/site-packages (from opt-einsum-fx>=0.1.4->e3nn==0.5.1->nablaDFT==2.0.0a0) (3.3.0)\n",
      "Requirement already satisfied: charset_normalizer<4,>=2 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from requests->torch-geometric==2.4.0->nablaDFT==2.0.0a0) (2.0.4)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from requests->torch-geometric==2.4.0->nablaDFT==2.0.0a0) (3.3)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from requests->torch-geometric==2.4.0->nablaDFT==2.0.0a0) (1.26.18)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from requests->torch-geometric==2.4.0->nablaDFT==2.0.0a0) (2021.10.8)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from jinja2->torch==2.2.0->nablaDFT==2.0.0a0) (2.0.1)\n",
      "Requirement already satisfied: joblib>=1.1.1 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from scikit-learn->torch-geometric==2.4.0->nablaDFT==2.0.0a0) (1.3.2)\n",
      "Requirement already satisfied: threadpoolctl>=2.0.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from scikit-learn->torch-geometric==2.4.0->nablaDFT==2.0.0a0) (2.2.0)\n",
      "Requirement already satisfied: absl-py>=0.4 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from tensorboard->schnetpack==2.0.4->nablaDFT==2.0.0a0) (1.2.0)\n",
      "Requirement already satisfied: grpcio>=1.48.2 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from tensorboard->schnetpack==2.0.4->nablaDFT==2.0.0a0) (1.51.3)\n",
      "Requirement already satisfied: google-auth<3,>=1.6.3 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from tensorboard->schnetpack==2.0.4->nablaDFT==2.0.0a0) (2.40.3)\n",
      "Requirement already satisfied: google-auth-oauthlib<0.5,>=0.4.1 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from tensorboard->schnetpack==2.0.4->nablaDFT==2.0.0a0) (0.4.6)\n",
      "Requirement already satisfied: markdown>=2.6.8 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from tensorboard->schnetpack==2.0.4->nablaDFT==2.0.0a0) (3.3.4)\n",
      "Requirement already satisfied: tensorboard-data-server<0.8.0,>=0.7.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from tensorboard->schnetpack==2.0.4->nablaDFT==2.0.0a0) (0.7.0)\n",
      "Requirement already satisfied: tensorboard-plugin-wit>=1.6.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from tensorboard->schnetpack==2.0.4->nablaDFT==2.0.0a0) (1.8.1)\n",
      "Requirement already satisfied: werkzeug>=1.0.1 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from tensorboard->schnetpack==2.0.4->nablaDFT==2.0.0a0) (2.0.3)\n",
      "Requirement already satisfied: wheel>=0.26 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from tensorboard->schnetpack==2.0.4->nablaDFT==2.0.0a0) (0.37.1)\n",
      "Requirement already satisfied: attrs>=17.3.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]>=2022.5.0->pytorch_lightning==2.1.4->nablaDFT==2.0.0a0) (25.3.0)\n",
      "Requirement already satisfied: multidict<7.0,>=4.5 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]>=2022.5.0->pytorch_lightning==2.1.4->nablaDFT==2.0.0a0) (5.2.0)\n",
      "Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]>=2022.5.0->pytorch_lightning==2.1.4->nablaDFT==2.0.0a0) (4.0.1)\n",
      "Requirement already satisfied: yarl<2.0,>=1.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]>=2022.5.0->pytorch_lightning==2.1.4->nablaDFT==2.0.0a0) (1.6.3)\n",
      "Requirement already satisfied: frozenlist>=1.1.1 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]>=2022.5.0->pytorch_lightning==2.1.4->nablaDFT==2.0.0a0) (1.2.0)\n",
      "Requirement already satisfied: aiosignal>=1.1.2 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]>=2022.5.0->pytorch_lightning==2.1.4->nablaDFT==2.0.0a0) (1.2.0)\n",
      "Requirement already satisfied: smmap<6,>=3.0.1 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from gitdb<5,>=4.0.1->GitPython!=3.1.29,>=1.0.0->wandb==0.16.3->nablaDFT==2.0.0a0) (5.0.1)\n",
      "Requirement already satisfied: cachetools<6.0,>=2.0.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from google-auth<3,>=1.6.3->tensorboard->schnetpack==2.0.4->nablaDFT==2.0.0a0) (5.5.2)\n",
      "Requirement already satisfied: pyasn1-modules>=0.2.1 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from google-auth<3,>=1.6.3->tensorboard->schnetpack==2.0.4->nablaDFT==2.0.0a0) (0.2.8)\n",
      "Requirement already satisfied: rsa<5,>=3.1.4 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from google-auth<3,>=1.6.3->tensorboard->schnetpack==2.0.4->nablaDFT==2.0.0a0) (4.7.2)\n",
      "Requirement already satisfied: requests-oauthlib>=0.7.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from google-auth-oauthlib<0.5,>=0.4.1->tensorboard->schnetpack==2.0.4->nablaDFT==2.0.0a0) (1.3.1)\n",
      "Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from pyasn1-modules>=0.2.1->google-auth<3,>=1.6.3->tensorboard->schnetpack==2.0.4->nablaDFT==2.0.0a0) (0.4.8)\n",
      "Requirement already satisfied: oauthlib>=3.0.0 in /home/kostanew/anaconda3/lib/python3.9/site-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<0.5,>=0.4.1->tensorboard->schnetpack==2.0.4->nablaDFT==2.0.0a0) (3.2.2)\n",
      "Downloading apsw-3.45.1.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (5.3 MB)\n",
      "   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.3/5.3 MB 5.1 MB/s eta 0:00:001\n",
      "\u001b[?25hUsing cached e3nn-0.5.1-py3-none-any.whl (118 kB)\n",
      "Downloading h5py-3.10.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.8 MB)\n",
      "   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.8/4.8 MB 18.4 MB/s eta 0:00:00\n",
      "\u001b[?25hDownloading hydra_core-1.2.0-py3-none-any.whl (151 kB)\n",
      "Using cached pytorch_lightning-2.1.4-py3-none-any.whl (778 kB)\n",
      "Using cached sympy-1.12-py3-none-any.whl (5.7 MB)\n",
      "Downloading torch-2.2.0-cp39-cp39-manylinux1_x86_64.whl (755.5 MB)\n",
      "   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 755.5/755.5 MB 27.9 MB/s eta 0:00:00\n",
      "\u001b[?25hUsing cached wandb-0.16.3-py3-none-any.whl (2.2 MB)\n",
      "Using cached nvidia_cublas_cu12-12.1.3.1-py3-none-manylinux1_x86_64.whl (410.6 MB)\n",
      "Using cached nvidia_cuda_cupti_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (14.1 MB)\n",
      "Using cached nvidia_cuda_nvrtc_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (23.7 MB)\n",
      "Using cached nvidia_cuda_runtime_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (823 kB)\n",
      "Using cached nvidia_cudnn_cu12-8.9.2.26-py3-none-manylinux1_x86_64.whl (731.7 MB)\n",
      "Using cached nvidia_cufft_cu12-11.0.2.54-py3-none-manylinux1_x86_64.whl (121.6 MB)\n",
      "Using cached nvidia_curand_cu12-10.3.2.106-py3-none-manylinux1_x86_64.whl (56.5 MB)\n",
      "Using cached nvidia_cusolver_cu12-11.4.5.107-py3-none-manylinux1_x86_64.whl (124.2 MB)\n",
      "Using cached nvidia_cusparse_cu12-12.1.0.106-py3-none-manylinux1_x86_64.whl (196.0 MB)\n",
      "Using cached nvidia_nccl_cu12-2.19.3-py3-none-manylinux1_x86_64.whl (166.0 MB)\n",
      "Using cached nvidia_nvtx_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (99 kB)\n",
      "Downloading triton-2.2.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (167.9 MB)\n",
      "   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 167.9/167.9 MB 32.1 MB/s eta 0:00:00\n",
      "\u001b[?25hDownloading opt_einsum_fx-0.1.4-py3-none-any.whl (13 kB)\n",
      "Downloading nvidia_nvjitlink_cu12-12.9.86-py3-none-manylinux2010_x86_64.manylinux_2_12_x86_64.whl (39.7 MB)\n",
      "   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 39.7/39.7 MB 32.9 MB/s eta 0:00:00\n",
      "\u001b[?25hBuilding wheels for collected packages: nablaDFT\n",
      "  Building wheel for nablaDFT (pyproject.toml) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for nablaDFT: filename=nabladft-2.0.0a0-py3-none-any.whl size=325691 sha256=f0a38c109467e50288d1d7abf119b0ff63238c256f92ef8b1c707f25551973e6\n",
      "  Stored in directory: /tmp/pip-ephem-wheel-cache-q5jffkaz/wheels/95/dc/4b/a7f373dfa8182ee05e77572c8d733599314dc911f94d056e09\n",
      "Successfully built nablaDFT\n",
      "Installing collected packages: triton, torch-scatter, sympy, nvidia-nvtx-cu12, nvidia-nvjitlink-cu12, nvidia-nccl-cu12, nvidia-curand-cu12, nvidia-cufft-cu12, nvidia-cuda-runtime-cu12, nvidia-cuda-nvrtc-cu12, nvidia-cuda-cupti-cu12, nvidia-cublas-cu12, h5py, apsw, torch-sparse, torch-cluster, nvidia-cusparse-cu12, nvidia-cudnn-cu12, hydra-core, wandb, nvidia-cusolver-cu12, torch, opt-einsum-fx, pytorch_lightning, e3nn, nablaDFT\n",
      "  Attempting uninstall: triton\n",
      "    Found existing installation: triton 2.0.0\n",
      "    Uninstalling triton-2.0.0:\n",
      "      Successfully uninstalled triton-2.0.0\n",
      "  Attempting uninstall: sympy\n",
      "    Found existing installation: sympy 1.10.1\n",
      "    Uninstalling sympy-1.10.1:\n",
      "      Successfully uninstalled sympy-1.10.1\n",
      "  Attempting uninstall: h5py\n",
      "    Found existing installation: h5py 3.6.0\n",
      "    Uninstalling h5py-3.6.0:\n",
      "      Successfully uninstalled h5py-3.6.0\n",
      "  Attempting uninstall: apsw\n",
      "    Found existing installation: apsw 3.44.0.0\n",
      "    Uninstalling apsw-3.44.0.0:\n",
      "      Successfully uninstalled apsw-3.44.0.0\n",
      "  Attempting uninstall: torch-cluster\n",
      "    Found existing installation: torch_cluster 1.6.3+pt25cpu\n",
      "    Uninstalling torch_cluster-1.6.3+pt25cpu:\n",
      "      Successfully uninstalled torch_cluster-1.6.3+pt25cpu\n",
      "  Attempting uninstall: hydra-core\n",
      "    Found existing installation: hydra-core 1.3.2\n",
      "    Uninstalling hydra-core-1.3.2:\n",
      "      Successfully uninstalled hydra-core-1.3.2\n",
      "  Attempting uninstall: wandb\n",
      "    Found existing installation: wandb 0.16.0\n",
      "    Uninstalling wandb-0.16.0:\n",
      "      Successfully uninstalled wandb-0.16.0\n",
      "  Attempting uninstall: torch\n",
      "    Found existing installation: torch 1.13.1\n",
      "    Uninstalling torch-1.13.1:\n",
      "      Successfully uninstalled torch-1.13.1\n",
      "  Attempting uninstall: pytorch_lightning\n",
      "    Found existing installation: pytorch-lightning 2.1.2\n",
      "    Uninstalling pytorch-lightning-2.1.2:\n",
      "      Successfully uninstalled pytorch-lightning-2.1.2\n",
      "  Attempting uninstall: nablaDFT\n",
      "    Found existing installation: nablaDFT 2.0.0\n",
      "    Uninstalling nablaDFT-2.0.0:\n",
      "      Successfully uninstalled nablaDFT-2.0.0\n",
      "ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "torchaudio 0.13.1 requires torch==1.13.1, but you have torch 2.2.0 which is incompatible.\n",
      "torchvision 0.14.1 requires torch==1.13.1, but you have torch 2.2.0 which is incompatible.\n",
      "Successfully installed apsw-3.45.1.0 e3nn-0.5.1 h5py-3.10.0 hydra-core-1.2.0 nablaDFT-2.0.0a0 nvidia-cublas-cu12-12.1.3.1 nvidia-cuda-cupti-cu12-12.1.105 nvidia-cuda-nvrtc-cu12-12.1.105 nvidia-cuda-runtime-cu12-12.1.105 nvidia-cudnn-cu12-8.9.2.26 nvidia-cufft-cu12-11.0.2.54 nvidia-curand-cu12-10.3.2.106 nvidia-cusolver-cu12-11.4.5.107 nvidia-cusparse-cu12-12.1.0.106 nvidia-nccl-cu12-2.19.3 nvidia-nvjitlink-cu12-12.9.86 nvidia-nvtx-cu12-12.1.105 opt-einsum-fx-0.1.4 pytorch_lightning-2.1.4 sympy-1.12 torch-2.2.0 torch-cluster-1.6.3+pt22cu121 torch-scatter-2.1.2+pt22cu121 torch-sparse-0.6.18+pt22cu121 triton-2.2.0 wandb-0.16.3\n",
      "\n",
      "[notice] A new release of pip is available: 24.3.1 -> 25.2\n",
      "[notice] To update, run: pip install --upgrade pip\n"
     ]
    }
   ],
   "source": [
    "!git clone https://github.com/AIRI-Institute/nablaDFT; cd nablaDFT/; pip install ."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0b642efa-0c23-4409-9b2e-7ea31c3dd355",
   "metadata": {},
   "source": [
    "# Загрузка данных"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "48889118-a332-4ce2-a6c3-986f672a1b3a",
   "metadata": {},
   "source": [
    "Этот код загружает метаинформацию для работы с nabla2DFT.\n",
    "Изначально, файл -- это csv-формат сжатый в архив.\n",
    "\n",
    "- `wget` -- скачивает файл\n",
    "- `gzip -d` -- распаковывает его"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "42c8d357-3f53-49d6-a76d-824a8af3d4a9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2025-09-08 21:38:32--  http://a002dlils-kadurin-nabladft.obs.ru-moscow-1.hc.sbercloud.ru/data/nablaDFTv2/summary.csv.gz\n",
      "Resolving a002dlils-kadurin-nabladft.obs.ru-moscow-1.hc.sbercloud.ru (a002dlils-kadurin-nabladft.obs.ru-moscow-1.hc.sbercloud.ru)... 46.243.206.34, 46.243.206.35\n",
      "Connecting to a002dlils-kadurin-nabladft.obs.ru-moscow-1.hc.sbercloud.ru (a002dlils-kadurin-nabladft.obs.ru-moscow-1.hc.sbercloud.ru)|46.243.206.34|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1997105126 (1,9G) [binary/octet-stream]\n",
      "Saving to: ‘summary.csv.gz’\n",
      "\n",
      "summary.csv.gz        4%[                    ]  82,50M  32,1MB/s               ^C\n",
      "gzip: summary.csv already exists; do you wish to overwrite (y or n)? ^C\n"
     ]
    }
   ],
   "source": [
    "!wget http://a002dlils-kadurin-nabladft.obs.ru-moscow-1.hc.sbercloud.ru/data/nablaDFTv2/summary.csv.gz\n",
    "!gzip -d summary.csv.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9b5b4105-88c8-4bcc-a8bc-9c97d3a7e2f5",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10 (p4env)",
   "language": "python",
   "name": "p4env"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
